cdxev/set.py

```diff
@@ -220,9 +220,15 @@ def run(sbom: dict, updates: t.Sequence[dict[str, t.Any]], cfg: SetConfig) -> No
     ctx.component_map = _map_out_components(sbom)
 
     for update in updates:
-        target_list: list[dict]
+        target_list: list[dict] = []
         try:
-            target_list = ctx.component_map[update["id"][0]]
+            update_id = update["id"]
+            if isinstance(update_id._keys[0].key, CoordinatesWithVersionRange):
+                for key in ctx.component_map.keys():
+                    if update_id._keys[0] == key:
+                        target_list += ctx.component_map[key]
+            else:
+                target_list = ctx.component_map[update["id"][0]]
```

Comment:
Two points:
- Do not access the private `_keys` attribute of the identity object. As a rule, anything that starts with a single or double underscore is private by convention and shouldn't be accessed from the outside. The class is designed as a collection so you can simply write `update_id[0]` as demonstrated in line 231.
- You're storing a reference in `update_id` but you haven't updated line 231 to make us of that same reference. This is not a functional problem, only a stylistic one. It makes it harder for readers to understand that both parts of the code actually reference the same object.  
  Either use the `update["id"][0]` syntax from ln 231 everywhere throughout this block or store a reference to the key (i.e., `update_key = update["id"][0]`) and apply it to every place where it is accessed, including ln 231.

Reply:
Thanks, adopted `update_key` for all occurences.